### ISSUE TITLE
x86_64-win32 build_engine failing

### ISSUE THREAD

**Issue by dlannan** [defold/defold#8144]
**Describe the bug (REQUIRED)**
When attempting to setup and build the engine on a Win11 PC (dual Xeon with 128GB ram and RTX2070) the build fails when executing this portion of the build process:
```./scripts/build.py build_engine --platform=x86_64-win32 --skip-tests -- --skip-build-tests```

**To Reproduce (REQUIRED)**
1. Following the build process here: https://github.com/defold/defold/blob/dev/README_BUILD.md#step-3---build-the-engine
2. The proicess stops with "FileNotFoundError: [WinError 2] The system cannot find the file specified" and "run.ExecException: 2"

**Expected behavior (REQUIRED)**
Build process to complete correctly?

**Defold version (REQUIRED):**
  - Version [e.g. 1.2.155]

**Platforms (REQUIRED):**
 - Platforms: Desktop PC Windows 11
 - OS: Windows 11
 - Device: NA

**Logs (OPTIONAL):**
Attached a log of the build process
[build_log.txt](https://github.com/defold/defold/files/12908401/build_log.txt)

**Additional context (OPTIONAL):**
I believe I followed all the setup documents correctly. It seems the failure is in the script trying to call clang++ which Im not sure if it should be doing this (I dont know). I have clang installed as part of the VS2019 install. 

Im running the build from a git bash shell setup with the appropriate paths etc.

**Comment by aglitchman** [comment]
I confirm that there is a bug. I think it was introduced after this commit - https://github.com/defold/defold/commit/d2d2c1990a5bcce7c617b7451c58a0c3d3fc2372

**Comment by dlannan** [comment]
Thanks @aglitchman (thought I had setup env etc wrong :) ). Will take a look at commit to see if I can move it forward.  If I can sort out a fix, I'll post a PR hopefully (time is a little tight atm).

**Comment by dlannan** [comment]
Ok. I think I have instructions to get past the issue and get a build running.

Firstl;y need to install scoop.
On windows, open a user powershel window (normal one not elevated) and type:
```
> Set-ExecutionPolicy RemoteSigned -Scope CurrentUser # Optional: Needed to run a remote script the first time
> irm get.scoop.sh | iex
```
This will install scoop. Reference: https://scoop.sh/

Then install llvm using scoop and I also reinstalled python using it too (just in case).  Using command:
```
> scoop install llvm
> scoop install python
```
Reference from the sokol github here: https://github.com/floooh/sokol/tree/master/bindgen

This process gets me a complete successful build. Thanks @aglitchman for the quick guidance. Looking forward to making a bunch of PR's :)

**Comment by JCash** [comment]
I'm still not sure what the actual problem was?
E.g. it builds on the [WIndows CI](https://github.com/defold/defold/actions), so I'd like to figure out what the actual problem was?

E.g. _which_ executable wasn't found?
```
  File "C:\Users\dlannan\AppData\Local\Programs\Python\Python310\lib\subprocess.py", line 1440, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
FileNotFoundError: [WinError 2] The system cannot find the file specified
```

**Comment by dlannan** [comment]
The problem was that clang wasnt accessible I think. I believe running scoop install fixed that for the sokol build gen system? Thats a best guess, but it did seem to resolve it. I had clang installed via VS2019, which maybe might work if I added certain paths etc? Dunno. I followed what was on the sokol github to resolve it :)

**Comment by dlannan** [comment]
Oop. sorry for closing again.

**Comment by dlannan** [comment]
> Yes, it makes sense, since we use clang to generate the AST info. I haven't double checked, but I assume clang arrives with visual studio nowadays (or is an easy step to add). We should make sure to add that as a note to the win32 build setup.

Yes you can install clang (I tried that by using VS Installer and modifying) but I think you need to add some paths for that to work properly (Im not sure). Running the scoop setup solved it for me tho, it also made sure I had recent python3 too. So Im happy with that process.

**Comment by JCash** [comment]
I believe we have clarified the need for Clang++ in our readme's, so I'm closing this now.
